src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -87,11 +103,12 @@ export class SavedObjectsClient {
 
     return {
       saved_objects: get(response, 'hits.hits', []).map(r => {
+        const docType =  get(r, '_source.type') || r._type;
         return {
```

Comment:
This might be an appropriate time to make a function to handle the returned object from these methods.